src/components/vm/consoles/consoles.jsx

```diff
@@ -88,7 +88,25 @@ class Consoles extends React.Component {
     onDesktopConsoleDownload (type) {
         const { vm } = this.props;
         // fire download of the .vv file
-        domainDesktopConsole({ name: vm.name, id: vm.id, connectionName: vm.connectionName, consoleDetail: vm.displays.find(display => display.type == type) });
+        const consoleDetail = vm.displays.find(display => display.type == type);
+
+        let address;
+        if (cockpit.transport.host == "localhost") {
+            const app = cockpit.transport.application();
+            if (app.startsWith("cockpit+=")) {
+                address = app.substr(9);
```

Comment:
These 2 added lines are not executed by any test.

Reply:
Hmm, our tests connect to 127.0.0.2:9091, from the PoV of the VM it's remote. But as far as I know this isn't about browser → ws, but ws → session, no? I.e. no "remote ssh" session?

Reply:
I am assuming that `cockpit.transport.host == "localhost")` means "local `cockpit-session` session", as opposed to "ssh session via 'Connect to:'" (bastion host).

Then I don't understand this `if`. What would the bit after "cockpit+=" be if not "localhost"? It appears to me that we would only have a cockpit+ "redirect" if the transport host is *not* localhost. When would that ever fire?